### PR TITLE
keccak: fix type definition

### PR DESCRIPTION
**Pull request by odanado**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/keccak/index.d.ts [hunk 1]
```diff
@@ -15,13 +15,15 @@ export class Keccak extends Transform {
         options: TransformOptions,
     );
     update(data: string | Buffer, encoding?: BufferEncoding): Keccak;
-    digest(encoding?: BufferEncoding): Buffer;
+    digest(): Buffer;
+    digest(encoding: BufferEncoding): string;
```
Comment (odanado) [inline types/keccak/index.d.ts:19]: https://github.com/cryptocoinjs/keccak/blob/a6b485faadce2f0cdce737a776aefff88d817719/lib/api/keccak.js#L56

File: types/keccak/index.d.ts [hunk 2]
```diff
@@ -15,13 +15,15 @@ export class Keccak extends Transform {
...
 }
 
 export class Shake extends Transform {
     constructor(rate: number, capacity: number, delimitedSuffix: number | null, options: TransformOptions);
     update(data: string | Buffer, encoding?: BufferEncoding): Shake;
-    squeeze(dataByteLength: number, encoding?: BufferEncoding): Buffer;
+    squeeze(dataByteLength: number): Buffer;
+    squeeze(dataByteLength: number, encoding: BufferEncoding): string;
```
Comment (peterblazejewicz) [inline types/keccak/index.d.ts:26]: similarly:
https://github.com/cryptocoinjs/keccak/blob/a6b485faadce2f0cdce737a776aefff88d817719/lib/api/shake.js#L51

**Review by peterblazejewicz — APPROVED** [review]
LGTM
@odanado thx for contribution!

**Comment by odanado** [comment]
Ready to merge

**Pull request merged**